### DISCUSSION TITLE
How to get line number of method in Interface

### DISCUSSION THREAD

**Discussion by huifengshi** [spotbugs/spotbugs#2949] [Q&A] [answered]
I need to analyze method  in Interface, when create a new BugInstance and addSourceLine，there is no source line data in org.apache.bcel.classfile.Method object. Is There any way to solve this problem, Thanks!

**Comment by welcome** [comment]
Thanks for opening your first issue here! :smiley:
Please check [our contributing guideline](https://github.com/spotbugs/spotbugs/blob/master/.github/CONTRIBUTING.md). Especially when you report a problem, make sure you share [a Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) to reproduce it in this issue.

**Comment by JuditKnoll** [comment, marked as answer]
I'm sorry for the belated answer. If you are writing a detector, you can get the sourceline data from the instruction, the pc, or the visitor itself. I suggest you to take a look at the usages of the `BugInstance.addSourceLine()` functions in the SpotBugs code, especially in the detectors which are in the [edu.umd.cs.findbugs.detect](https://github.com/spotbugs/spotbugs/tree/master/spotbugs/src/main/java/edu/umd/cs/findbugs/detect) package.
If you want to refer the entire method, not a specific line, you can use one of the [SourceLineAnnotation.forEntireMethod()](https://github.com/spotbugs/spotbugs/blob/master/spotbugs/src/main/java/edu/umd/cs/findbugs/SourceLineAnnotation.java#L285) methods.
